apigee_edge.module

```diff
@@ -81,6 +82,9 @@ function apigee_edge_module_implements_alter(&$implementations, $hook) {
  */
 function apigee_edge_theme() {
   return [
+    'apigee_entity_default' => [
```

Comment:
Having one template for all entity types is weird. 

I see what you're doing here but I think we should have separate templates for entity types. 

Example: `developer_app`,  `team_app`, `api_product`.

Reply:
See my comment above.
If you still disagree please describe what we would win by defining the entity type specific themes/temples with the same content? (It would mean that the MINT module also must define its own entity type specific themes and templates as well instead of using the EdgeEntityViewBuilder as a view builder and automatically using the fallback theme "apigee_edge_default" if no customization is needed in the HTML markup.)

Reply:
Can we please KISS and implement each entity with its own named hook, and template? It would help toward solving a lot of issues happening in this PR, and make the end result easier for anyone working with it to understand. 

The first thing someone looking to customize one of these templates will do is look for the a template named after the entity. Let's just give them that.  It's well worth having multiple templates, even if they all have the same, basic markup because this is about enabling the _ability_ to customize these entities just like anyone doing theme development would for any other entity: node, media, paragraph, or otherwise. Anything else is an anti-pattern that is bound to confuse people. 

Another negative side effect of doing it this way is `apigee_entity_default` is the hook that whole pipeline is affected (preprocess, suggestions, etc), and will have to be dealt with through this prefix, as noted in the comments of the template.  That's a terrible TX to put onto theme developers, and we shouldn't do it.

Reply:
I do agree with @jacine on the fact that our goal is to make things simple for developer that need to make customizations even if we have some basic template markup repetition.  Let’s error on the side of ease of use than worrying about a bit of replication in templates

Reply:
@jacine Please read my explanation in the next comment which I hope also explains what do I mean when this way of doing this is KISS.
https://github.com/apigee/apigee-edge-drupal/pull/181#discussion_r278981143

Drupal core already provides a way to KISS without defining (copy-pasting) theme for all entities that you have. This is implementing a theme hook with the entity_type.

If this [patch](https://www.drupal.org/project/drupal/issues/2808481#comment-11775158) lands in core it will also try to suggest not to implement themes for all entities you have because Drupal will provide a default `entity` theme. So if will you need small customization you can do it by using the new `entity--ENTITY_TYPE.html.twig` template. If you need something more custom, something that always override the default entity theme, you should implement an entity type-specific theme hook and do your stuff there. If there is an entity type specific defined it will always win. This is why I am trying to suggest not to implement entity type specific themes in this module, leave it to the themers to do that in their theme/custom module and they will have full control over everything. Which I guess even better than fighting against defaults that a module provides/enforces.
This is how I understand what is happening in 2808481.

Reply:
> The first thing someone looking to customize one of these templates will do is look for the a template named after the entity.
>  this is about enabling the ability to customize these entities just like anyone doing theme development would for any other entity: node, media, paragraph, or otherwise

Probably these entities only have default themes because https://www.drupal.org/project/drupal/issues/2808481  has not been fixed yet.

@Jaesin Please also FIXME if I am making the wrong assumption, but as a back-ender I think this is how a front-ender's mind works, when someone says you need to customize the developer_app/team/team_app page:
* As you wrote: look for the a template named after the entity.
* If there is no template, enable the theme debug and check what template is being applied on the page
* It is the apigee-entity.html.twig, no template suggestions other than that.
* Ok, let's check what it is in that template:
https://github.com/apigee/apigee-edge-drupal/blob/02a29d6f5082306bfa319eca57c5c157962ead1a/templates/apigee-entity-default.html.twig#L9-L31
* Cool, so I'll define my own template, I can also re-use as much I want, but my hands are not tied by default theme that the entity provides.